finesse/hardware/dp9800.py

```diff
@@ -0,0 +1,263 @@
+"""This module provides an interface to DP9800 temperature readers."""
+import logging
+from datetime import datetime
+from decimal import Decimal
+
+from pubsub import pub
+from serial import EIGHTBITS, PARITY_NONE, STOPBITS_ONE, Serial, SerialException
+
+
+class DP9800Error(Exception):
+    """Indicates that an error occurred while communicating with the device."""
+
+
+class DP9800:
+    """An interface for DP9800 temperature readers.
+
+    The manual for this device is available at:
+    https://assets.omega.com/manuals/M5210.pdf
+    """
+
+    NUM_CHANNELS = 8
+    """The number of channels on the DP9800 device."""
```

Comment:
I don't think you need to hardcode it. Just parse however many temperatures the device gives you in a given reading.